src/utils.cairo

```diff
@@ -1,6 +1,94 @@
-use core::sha256::{compute_sha256_byte_array, compute_sha256_u32_array};
+use core::fmt::{Display, Formatter, Error};
 use core::num::traits::{Zero, One, BitSize};
-use core::starknet::secp256_trait::Secp256PointTrait;
+use core::sha256::{compute_sha256_byte_array, compute_sha256_u32_array};
+use core::to_byte_array::AppendFormattedToByteArray;
+
+#[derive(Copy, Drop, Debug)]
+// pub is required here as we define Hash in utils and we need to import Hash in merkle_tree.cairo
+// for arguments to merkle_root function
+pub struct Hash {
+    value: [u32; 8]
+}
+
+#[generate_trait]
+pub impl HashImpl of HashTrait {
+    #[inline(always)]
+    fn to_hash(array: [u32; 8]) -> Hash {
+        Hash { value: array }
+    }
+}
+
+impl HashDisplay of Display<Hash> {
+    fn fmt(self: @Hash, ref f: Formatter) -> Result<(), Error> {
+        let hash: u256 = (*self).into();
+        let base: u256 = 16;
+
+        hash.append_formatted_to_byte_array(ref f.buffer, base.try_into().unwrap());
```

Comment:
why not:
```
hash.append_formatted_to_byte_array(ref f.buffer, 16);
```
?

Reply:
updated.
Thought i had to define a `NonZero<u256` like i did, but indeed it works this way. thanks!